### DISCUSSION TITLE
gpt-4 does it work for you? Or does it always go to gpt-4-0613?

### DISCUSSION THREAD

**Discussion by githubdebugger** [danny-avila/LibreChat#869] [Q&A] [answered]
I have this line set in the .env:

"OPENAI_MODELS=gpt-3.5-turbo-0613,gpt-3.5-turbo-16k-0613,gpt-4,gpt-4-0314,gpt-4-0613"

However, even when I chose the plain old vanilla/original gpt-4, I always see gpt-4-0613 under usage, if I select gpt-4-0314, it does show the API endpoint as gpt-4-0314. I had my doubts as the original gpt-4 was quite slow vs 0314 or 0613, but the responses that I was getting were fast.

I do have access to old gpt-4 API as it shows this under rate limits:
gpt-4 | 40,000 | 200
gpt-4-0314 | 40,000 | 200
gpt-4-0613 | 40,000 | 200

Is it just me or this is how it is for you guys as well? OR if there are some issues with my settings in the .env for OPENAI_MODELS?

Thanks!

**Comment by danny-avila** [comment, marked as answer]
As of late June, `gpt-4-0613` is the default model when using `gpt-4`. I can't remember where I read this, but this is definitely the case.

`gpt-4-0314` will deprecate next year.

**Reply by githubdebugger** [reply]
We do not have any ways to select the original GPT-4 model then? 

I remember reading somewhere (few months back) that OpenAI will keep the original model to be accessed via API until next year.

**Comment by jamesrandall007** [comment]
For some reason gpt-4 doesn't work at all, it always replies "I'm gpt-3". I pay the additional to have  access through the api. Don't really know where to start  debugging.

**Reply by danny-avila** [reply]
It’s gpt 4. It likely was trained on some gpt 3 synthetic data so it will say that

**Comment by jamesrandall007** [comment]
It didn't even occur to me that might be the case. Thank you!


On Thu, Sep 7, 2023 at 1:39 PM Danny Avila ***@***.***> wrote:

> It’s gpt 4. It likely was trained on some gpt 3 synthetic data so it will
> say that
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danny-avila/LibreChat/discussions/869#discussioncomment-6941434>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQMRLT7O3ZDZNQRSFBR4RLDXZIWGZANCNFSM6AAAAAA4H6JVEM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
James Randall
COO | Kassay Family Medicine
phone: 503-675-1137 x5 <http://5036751137+5>

site: KaraKassay.com <http://karakassay.com/>
email: ***@***.***
address: 12511 SW 68th Ave Portland, OR 97223
[image: facebook] <https://www.facebook.com/KassayFamilyMedicine/>

Please note that any view of the opinions expressed in this e-mail is
solely those of the author and do not necessarily represent those of its
affiliates. If you are not the intended recipient(s), or agent responsible
to deliver this message to the intended recipient(s), you are hereby
notified that any review, e-transmission, conversion to hard copy, copying,
circulation or other use of this message is strictly prohibited and may be
illegal.

**Reply by githubdebugger** [reply]
Did you try on playground on the OpenAI website by selecting GPT-4?

**Comment by jamesrandall007** [comment]
directly on website queryk specify gpt-4,
```
"What version are you?"
```
ChatGPT
```
I am based on the GPT-4 architecture, which is a more advanced version
compared to the GPT-3 model. My training included information up until
September 2021, so my knowledge goes up until that point. If you have any
questions or need information available until 2021, feel free to ask!
```
Same question in playground with 3.5
```
I am based on the GPT-3.5 architecture, and my knowledge is up to date only
until September 2021. If you have any questions or need assistance with
programming, Python, web development, or any other topic, feel free to ask,
and I'll do my best to help you.
```
There is an error in console log right after submitting:
```
onloadwff.js:71 Assertion failed: Input argument is not an HTMLInputElement
getFormProfile @ onloadwff.js:71
setFieldValue @ onloadwff.js:71
formKeydownListener @ onloadwff.js:71
```
you can see the request is going out at gpt-4
```

   1. *conversation**: **{conversationId: null, title: 'New
   Chat', endpoint: 'openAI', createdAt: '', updatedAt: '', …}*
   2. *endpointOption**: *
   *{endpoint: 'openAI', model: 'gpt-4', chatGptLabel: null,
promptPrefix: null, temperature: 1, …}*
   3. *initialResponse**: **{sender: 'ChatGPT', text: '<span
   className="result-streaming">█</span>', parentMessageId:
'ab78106f-f037-4a17-a8cb-XXXX', messageId:
'ab78106f-f037-4a17-a8cb-XXXX', conversationId: null, …}*
   4. *isRegenerate**: **false*
   5. *message**: **{sender: 'User', text: 'what gpt model are
   you', current: true, isCreatedByUser: true, parentMessageId:
'00000000-0000-0000-0000-000000000000', …}*
   6. *messages**: **[]*
   7.
8. *```*


On Fri, Sep 8, 2023 at 3:31 AM githubdebugger ***@***.***>
wrote:

> Did you try on playground on the OpenAI website by selecting GPT-4?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danny-avila/LibreChat/discussions/869#discussioncomment-6946923>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQMRLT2LDEVQIJGHVOBQLTLXZLXY5ANCNFSM6AAAAAA4H6JVEM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
James Randall
COO | Kassay Family Medicine
phone: 503-675-1137 x5 <http://5036751137+5>

site: KaraKassay.com <http://karakassay.com/>
email: ***@***.***
address: 12511 SW 68th Ave Portland, OR 97223
[image: facebook] <https://www.facebook.com/KassayFamilyMedicine/>

Please note that any view of the opinions expressed in this e-mail is
solely those of the author and do not necessarily represent those of its
affiliates. If you are not the intended recipient(s), or agent responsible
to deliver this message to the intended recipient(s), you are hereby
notified that any review, e-transmission, conversion to hard copy, copying,
circulation or other use of this message is strictly prohibited and may be
illegal.
